### PR TITLE
Feature/ndev 2104 new debug methods

### PR DESCRIPTION
**Pull request by romanova-natasha**

### CODE REVIEW

File: integration/tests/tracer/test_tracer_debug_methods.py [hunk 1]
```diff
@@ -254,4 +255,164 @@ def test_debug_trace_block_by_non_existent_hash(self):
...
+            self.sender_account, self.recipient_account, 0.1)
+        assert receipt["status"] == 1
+        wait_condition(lambda: self.tracer_api.send_rpc(method="debug_getRawHeader",
+                                                        params=[hex(receipt["blockNumber"])])["result"] is not None,
+                       timeout_sec=120)
+        
+        response = self.tracer_api.send_rpc(
+            method="debug_getRawHeader", params=[hex(receipt["blockNumber"])])
+        assert "error" not in response, "Error in response"
+        assert response["result"] is not None and response["result"] != ""
```
Comment (gigimon) [inline integration/tests/tracer/test_tracer_debug_methods.py]: will be goood check some data inside response

File: integration/tests/tracer/test_tracer_debug_methods.py [hunk 2]
```diff
@@ -254,4 +255,164 @@ def test_debug_trace_block_by_non_existent_hash(self):
...
+            self.sender_account, self.recipient_account, 0.1)
+        assert receipt["status"] == 1
+        wait_condition(lambda: self.tracer_api.send_rpc(method="debug_getRawHeader",
+                                                        params=[receipt["blockHash"].hex()])["result"] is not None,
+                       timeout_sec=120)
+        
+        response = self.tracer_api.send_rpc(
+            method="debug_getRawHeader", params=[receipt["blockHash"].hex()])
+        assert "error" not in response, "Error in response"
+        assert response["result"] is not None and response["result"] != ""
```
Comment (gigimon) [inline integration/tests/tracer/test_tracer_debug_methods.py]: check body too. Because not blank response is not verify that response work good

**Review by gigimon — APPROVED** [review]
pls check some data inside responses. Will be good to check scheme for all responses. Maybe in next issue we could add schemes and use jsonschema package to verify

File: integration/tests/tracer/test_tracer_debug_methods.py (hunk 1 above)
Comment (romanova-natasha) [reply integration/tests/tracer/test_tracer_debug_methods.py]: done

File: integration/tests/tracer/test_tracer_debug_methods.py (hunk 2 above)
Comment (romanova-natasha) [reply integration/tests/tracer/test_tracer_debug_methods.py]: done

File: integration/tests/tracer/test_tracer_debug_methods.py [hunk 3]
```diff
@@ -254,4 +257,183 @@ def test_debug_trace_block_by_non_existent_hash(self):
...
+        wait_condition(lambda: self.tracer_api.send_rpc(method="debug_getModifiedAccountsByNumber",
+                                                        params=[start_number, end_number])["result"] is not None,
+                       timeout_sec=120)
+        
+        response = self.tracer_api.send_rpc(
+            method="debug_getModifiedAccountsByNumber", params=[start_number, end_number])
+        self.check_modified_accounts_response(response)
+    
+    @pytest.mark.skip(reason="bug NDEV-2375")
+    def test_debug_get_modified_accounts_by_number_201_blocks_difference(self):
```
Comment (kristinaNikolaevaa) [inline integration/tests/tracer/test_tracer_debug_methods.py:393]: why 201 ?

**Review by kristinaNikolaevaa — APPROVED** [review]

File: integration/tests/tracer/test_tracer_debug_methods.py (hunk 3 above)
Comment (romanova-natasha) [reply integration/tests/tracer/test_tracer_debug_methods.py:393]: cause allowed range of blocks is 200
it is negative case

**Pull request merged**